File: 프로그래머스/lv2/42577. 전화번호 목록/전화번호 목록.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

bool solution(vector<string> v) {
    sort(v.begin(), v.end());
    for (int i = 0; i < v.size() - 1; i++)
        if (v[i] == v[i + 1].substr(0, v[i].size())) return false;
    return true;
}

File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.java
import java.util.*;

class Solution {
    public boolean solution(String[] arr) {
        boolean answer = true;
        
        Arrays.sort(arr);

        for (int i = 0; i < arr.length - 1; i++) {
            if (arr[i + 1].startsWith(arr[i])) return false;
        }
        
        return answer;
    }
}